library/debug.c

```diff
@@ -194,6 +194,52 @@ void mbedtls_debug_print_ecp(const mbedtls_ssl_context *ssl, int level,
 }
 #endif /* MBEDTLS_ECP_LIGHT */
 
+#if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
+void mbedtls_debug_print_psa_ec(const mbedtls_ssl_context *ssl, int level,
+                                const char *file, int line,
+                                const char *text, const mbedtls_pk_context *pk)
+{
+    char str[DEBUG_BUF_SIZE];
+    mbedtls_mpi mpi;
+    const uint8_t *mpi_start;
+    size_t mpi_len;
+    int ret;
+
+    if (NULL == ssl              ||
+        NULL == ssl->conf        ||
+        NULL == ssl->conf->f_dbg ||
+        level > debug_threshold) {
+        return;
+    }
+
+    /* For the description of pk->pk_raw content please refer to the description
+     * psa_export_public_key() function. */
+    mpi_len = (pk->pub_raw_len - 1)/2;
+
+    /* X coordinate */
+    mbedtls_mpi_init(&mpi);
+    mpi_start = pk->pub_raw + 1;
+    ret = mbedtls_mpi_read_binary(&mpi, mpi_start, mpi_len);
```

Comment:
Yes, I noticed that there was the `mbedtls_debug_print_buf()` which seemed to do what I was looking for, however that would have required a change in the test case because the output is not 100% identical.

> 2d EPIC when we try to make things work without BIGNUM_C

Oh wow, thanks to this comment I just realized that there is an extra list of EPICs that I was not aware of ':)

Reply:
Yeah, it's easy to miss as it's on another board (we have board for active EPIC and another one for the future), but I added a link from the first card in the 2c EPIC.

Currently it's not really populated (just 1 issue that you already resolved some time ago), and with a bit of luck it will turn out to be trivial, considering the way we're handling things in the 2c EPIC. Once we have PK working without `ECP_LIGHT` (which should be fairly soon now: this PR, its private counterpart, then probably just 1 more PR I hope), I'll re-assess the situation and see what dependencies on `BIGNUM_C` remain. I'm hoping at that point it will be just `debug.c`, which as we were saying will probably be quite easy to handle.